processor/schemaprocessor/internal/translation/modifier.go

```diff
@@ -0,0 +1,194 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translation // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/translation"
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/alias"
+)
+
+// Modifier abstracts a verion change that allows
+// for the changes to be updated or revert.
+type Modifier interface {
+	// UpdateAttrs modifies all matching attributes
+	// names with the configured changes.
+	UpdateAttrs(attrs pcommon.Map)
```

Comment:
Hard to understand the diff between update and revert. After reading only this interface comments I don't know when to use one or the other, this is a bad sign in my opinion.

Reply:
I wonder if it is worth while writing it as:

```golang
type VersionIterator interface {
    NextAttributes(...)
    PrevAttributes(...)
}
```

Would that help make it clearer?